cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/upgrade/TemplateSetUpgraderTest.java

```diff
@@ -37,121 +37,122 @@
-				.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
-		ContextConfigurationUpgrader upgrader = new ContextConfigurationUpgrader();
-		upgrader.upgradeConfigurationToLatestVersion(contextLocation, BackupPolicy.ENFORCE_BACKUP);
-
-		Path newTemplatesPath = this.templateLocation.getParent().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
-		newTemplatesPath = newTemplatesPath.resolve(ConfigurationConstants.ADAPTED_FOLDER);
-
-		SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
-		try (InputStream schemaStream = getClass().getResourceAsStream("/schema/v3.0/contextConfiguration.xsd")) {
-			StreamSource schemaSourceStream = new StreamSource(schemaStream);
-			Schema schema = schemaFactory.newSchema(schemaSourceStream);
-			Validator validator = schema.newValidator();
-
-			for (String s : newTemplatesPath.toFile().list()) {
-				Path newContextPath = newTemplatesPath.resolve(s + "/" + "src/main/resources");
-				newContextPath = newContextPath.resolve("context.xml");
-				assertTrue(newContextPath.toFile().exists());
-				StreamSource contextStream = new StreamSource(newContextPath.toFile());
-				try {
-					validator.validate(contextStream);
-				} catch (SAXException e) {
-					fail("Exception show that validator has found an fault");
-					contextStream.getInputStream().close();
-					schemaStream.close();
-				}
-
-			}
-			schemaStream.close();
-		}
-
-	}
+  /** Root path to all resources used in this test case */
+  private static String testFileRootPath = "src/test/resources/testdata/unittest/config/upgrade/TemplateSetUpgraderTest/";
+
+  /** Path to the template folder */
+  private Path templateLocation;
+
+  /** JUnit Rule to create and automatically cleanup temporarily files/folders */
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Before
+  public void prepare() throws IOException {
+
+    Path playground = this.tempFolder.newFolder(".cobigen").toPath();
+    FileUtils.copyDirectory(new File(testFileRootPath + "valid-2.1"), playground.toFile());
+    this.templateLocation = playground.resolve(ConfigurationConstants.TEMPLATES_FOLDER);
+  }
+
+  /**
+   * Test the correct folder creation
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testTemplateSetUpgrade() throws Exception {
+
+    TemplateSetUpgrader templateSetUpgrader = new TemplateSetUpgrader();
+    templateSetUpgrader.upgradeTemplatesToTemplateSets(this.templateLocation);
+
+    Path templateSetsPath = this.templateLocation.getParent().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    Path templateSetsAdapted = templateSetsPath.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+    assertThat(templateSetsPath).exists();
+    assertThat(templateSetsAdapted).exists();
+
+  }
+
+  /**
+   * Tests if the Template files are copied correct into the new template set and into the backup folder
```

Comment:
Better: Tests if the template files were copied correctly into the new template set and the backup folder